### DISCUSSION TITLE
Sauvegarde vues explorations, ID technique et gestion des droits plus fine

### DISCUSSION THREAD

**Discussion by Blemaire-ssi** [dbarzin/mercator#763] [Ideas]
Bonsoir,

Tout d'abord, un grand bravo pour cet outil accessible qui nous permet aujourd'hui d'avancer sur nos travaux de cartographie quand les seuls moyens que nous avons à disposition sont humains sans support financier.

Au cours de notre utilisation, plusieurs fonctionnalités nous ont semblé avoir un intérêt :

- Possibilité de sauvegarder les vues faites lors de l'exploration pour pouvoir les consulter plus tard.
- Ajout d'un id technique dans chaque table accessible par l'API pour pouvoir y renseigner l'ID technique de la solution d'où proviennent les données (ex: quand j'importe en automatique mes VMs de mon outil d'inventaire, s'il y a renommage de la VM, ça en créera une nouvelle vu que nous n'avons pas d'identifiant commun aux 2 systèmes (à moins de maintenir une base de relation intermédiaire lors de l'import).
- Pouvoir rattacher les utilisateurs de la plateforme à une entité qui a été définie dans la vue de l'écosystème
- Dans la vue métier, pouvoir Limiter l'accès en lecture /écriture aux processus (et tous les éléments qui peuvent le composer) aux utilisateurs qui sont rattaché à l'entité qui gère ces processus

Peut être que les 2 derniers points que j'ai mentionné correspondent aux points suivants de la roadmap ?
- Généraliser la notion de cartographe à d'autres objets (cf.: https://laravel.com/docs/10.x/authorization)
- Générer les cartographes dans la gestion des utilisateurs

Nous avons aussi rencontré un bug lors de l'exploration : certains flux apparaissent en double quand on déploie sur plusieurs niveaux.

Bonne fin de soirée,

**Comment by dbarzin** [comment]
Voici la réponse aux questions :

- Possibilité de sauvegarder les vues faites lors de l'exploration pour pouvoir les consulter plus tard.

Le composant utilisé ne permet pas la sauvegarde, mais je travaille sur un autre composant qui permettra de sauvegarder les vues mais également d'ajouter des commentaires, des signes, de mettre à jour automatiquement les objets. Ce n'est pas prévu pour tout de suite mais dans le courant de l'année prochaine.

- Ajout d'un id technique dans chaque table accessible par l'API pour pouvoir y renseigner l'ID technique de la solution d'où proviennent les données (ex: quand j'importe en automatique mes VMs de mon outil d'inventaire, s'il y a renommage de la VM, ça en créera une nouvelle vu que nous n'avons pas d'identifiant commun aux 2 systèmes (à moins de maintenir une base de relation intermédiaire lors de l'import).

Effectivement, j'ai le même problème. Peut-être faudrait-il éviter de renommer les VM sans le faire pareillement dans Mercator.

Pour ce qui concerne ces questions qui se rapportent à la délégation de la cartographie aux utilisateurs : 

- Pouvoir rattacher les utilisateurs de la plateforme à une entité qui a été définie dans la vue de l'écosystème
- Dans la vue métier, pouvoir limiter l'accès en lecture /écriture aux processus (et tous les éléments qui peuvent le composer) aux utilisateurs qui sont rattachés à l'entité qui gère ces processus
- Généraliser la notion de cartographe à d'autres objets (cf. : https://laravel.com/docs/10.x/authorization)
- Générer les cartographes dans la gestion des utilisateurs

Je ne sais pas si c'est dans cette direction qu'il faut aller. Cela complexifie fortement la gestion des droits d'accès et je ne vois pas comment gérer la création de nouveaux objets.

Concernant le bug, pourrais-tu décrire le problème et créer une "issue" ?

Désolé pour la réponse tardive.

**Reply by Blemaire-ssi** [reply]
Merci pour toutes ces réponses, 

Super pour les nouvelles fonctionnalités touchant les vues, on attend de voir ça.

Pour le renommage des machines, ça n'arrive pas tous les jours donc nous faisons les corrections côté Mercator manuellement.  

A propos de la gestion des droits, je comprends la complexité que cela peut apporter et en ayant avancé sur notre projet cartographie, on se rend compte que l'absence de cette fonctionnalité ne pose aucun problème.

Pour finir sur le bug, je crée une "issue".
